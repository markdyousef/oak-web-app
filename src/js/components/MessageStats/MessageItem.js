// @flow
import React, { PropTypes } from 'react';
import moment from 'moment';

import css from './MessageStats.css';

const MessageItem = ({ message, onClick }:Object) => {
    return (
        <div
            className={css.messageItem}
            onClick={onClick}
        >
            <div className={css.userAvatar}>
                <span />
            </div>
            <div className={css.text}>
                <p>{message.text}</p>
            </div>
            <div className={css.time}>
                {moment(message.ts * 1000).fromNow()}
            </div>
        </div>
    );
}

MessageItem.propTypes = {
    message: PropTypes.shape({
        text: PropTypes.string.isRequired,
        user: PropTypes.string.isRequired,
        ts: PropTypes.string.isRequired
    }),
    onClick: PropTypes.func.isRequired
}

export default MessageItem;
