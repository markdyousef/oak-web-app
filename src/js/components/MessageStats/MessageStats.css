:root {
    --main-black-color: #000;
    --main-white-color: #fefeff;
    --main-orange-color: #ff9200;
    --main-red-color: #8b0000;
    --main-green-color: #005d6c;
}

.container {
    width: 100%;
    height: 400px;
    background-color: #fff;
    display: flex;
}

.messageList {
    width: 30%;
    height: 100%;
    overflow-x: scroll;
    background-color: var(--main-white-color);
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.messageItem {
    cursor: pointer;
    padding: 10px;
    height: 100px;
    display: flex;
    align-items: center;
    margin: 5px;
    border-radius: 4px;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
    background-color: #fff;
}

.messageItem:hover {
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
}

.userAvatar {
    height: 50px;
    width: 50px;
    background-color: var(--main-green-color);
    border-radius: 100px;
}

.text p {
    font-size: 10px;
    padding: 10px;
    width: 200px;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}

.time {
    margin-left: 10px;
    color: var(--main-black-color);
    font-size: 14px;
}

.messageStats {
    width: 70%;
}
