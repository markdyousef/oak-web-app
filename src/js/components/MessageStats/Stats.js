import React, { PropTypes } from 'react';

import css from './MessageStats.css';

const Stats = ({ message }: Object) => {
    return (
        <div className={css.stats}>
            {message.text}
        </div>
    );
};

Stats.propTypes = {
    message: PropTypes.shape({
        text: PropTypes.string.isRequired
    })
};

export default Stats;
