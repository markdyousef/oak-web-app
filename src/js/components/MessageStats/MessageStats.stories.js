import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import MessageStats from './MessageStats';
import MessageItem from './MessageItem';

storiesOf('MessageStats', module)
    .add('empty', () => (
        <MessageStats />
    ));
storiesOf('MessageItem', module)
    .add('MessageItem with text and time', () => (
        <MessageItem message={{ text: 'Hello', ts: 'now' }} onClick={action('clicked')} />
    ));
