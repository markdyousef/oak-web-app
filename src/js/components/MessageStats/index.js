import MessageStats from './MessageStats';

export default MessageStats;
