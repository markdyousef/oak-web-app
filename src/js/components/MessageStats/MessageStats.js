// @flow
import React, { Component, PropTypes } from 'react';
import MessageItem from './MessageItem';
import Stats from './Stats';

import css from './MessageStats.css';

class MessageStats extends Component {
    static propTypes = {
        data: PropTypes.shape({
            channel: PropTypes.string.isRequired,
            messages: PropTypes.arrayOf(PropTypes.object)
        })
    };
    constructor() {
        super();
        this.renderMessages = this.renderMessages.bind(this);
        this.renderStats = this.renderStats.bind(this);
        this.state = {
            activeMessage: null
        };
    }
    renderMessages() {
        const { data } = this.props;
        if (data && data.messages) {
            return data.messages.map(message => (
                <MessageItem
                    onClick={() => this.setState({ activeMessage: message })}
                    message={message}
                    key={message.ts}
                />
            ));
        }
        return <div className={css.noMessage}>No Messages</div>;
    }
    renderStats() {
        const { activeMessage } = this.state;

        if (activeMessage) {
            return <Stats message={activeMessage} />
        }
        return <div className={css.noStats}>No stats</div>
    }
    render() {
        return (
            <div className={css.container}>
                <div className={css.messageList}>
                    {this.renderMessages()}
                </div>
                <div className={css.messageStats}>
                    {this.renderStats()}
                </div>
            </div>
        );
    }
}

export default MessageStats;
