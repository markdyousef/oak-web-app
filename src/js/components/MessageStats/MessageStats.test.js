// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import MessageStats from './MessageStats';

test('renders MessageStats', () => {
    const props = {
    };
    const component = renderer.create(
        <MessageStats {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
