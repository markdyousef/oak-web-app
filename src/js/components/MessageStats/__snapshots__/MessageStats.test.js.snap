exports[`test renders MessageStats 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <div
      className={undefined}>
      No Messages
    </div>
  </div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      No stats
    </div>
  </div>
</div>
`;
